frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -214,10 +214,13 @@ fn construct_runtime_final_expansion(
 	let ExplicitRuntimeDeclaration {
 		name,
 		where_section: WhereSection { block, node_block, unchecked_extrinsic },
-		pallets,
+		mut pallets,
 		pallets_token,
 	} = definition;
 
+	// Ensure that the code is generated based on the pallet index ordering:
+	pallets.sort_by_key(|p| p.index);
```

Comment:
This will need very careful checking with existing runtimes, since it can completely brick the chain by changing the initialization order.

Reply:
The current situation seems unsatisfactory as it's ordering some things one way and others another. Do we need an opt in / opt out mechanism? Are there any plans of a construct_runtime v2 like we have recently with the benchmarks v2?